src/RootSolvers.jl

```diff
@@ -273,35 +544,415 @@ on ``|(x2-x1)/x1| || |x2-x1|``
+        s_is_between = ifelse(a < b, s > m && s < b, s < m && s > b)
+        
+        # If step is not acceptable, fall back to bisection
+        use_bisection = !s_is_between || abs(s - b) >= abs(e / 2) || abs(e) < eps(FT)
+        
+        s_new = ifelse(use_bisection, (a + b) / 2, s)
+        d_new = s_new - b
+        e_new = ifelse(use_bisection, d_new, d)
+        s, d, e = s_new, d_new, e_new
+
+        fs = f(s)
+        push_history!(x_history, s, soltype)
+        push_history!(y_history, fs, soltype)
+
+        c, fc = b, fb
+
+        # Update bracket
+        cond_bracket = fa * fs < 0
+        a_new = ifelse(cond_bracket, a, s)
+        fa_new = ifelse(cond_bracket, fa, fs)
+        b_new = ifelse(cond_bracket, s, b)
+        fb_new = ifelse(cond_bracket, fs, fb)
+        a, fa, b, fb = a_new, fa_new, b_new, fb_new
+
+        # Ensure abs(fb) <= abs(fa)
+        cond_swap = abs(fa) < abs(fb)
+        a_new_swap = ifelse(cond_swap, b, a)
+        b_new_swap = ifelse(cond_swap, a, b)
+        fa_new_swap = ifelse(cond_swap, fb, fa)
+        fb_new_swap = ifelse(cond_swap, fa, fb)
+        a, fa, b, fb = a_new_swap, fa_new_swap, b_new_swap, fb_new_swap
+    end
+
+    return SolutionResults(soltype, b, false, fb, maxiters, x_history, y_history)
+end
+
+# Dedicated helper for the Secant method (an open, two-point method)
+function _find_zero_secant(f, x0, x1, soltype, tol, maxiters)
+    FT = typeof(x0)
+    if !isfinite(x0) || !isfinite(x1)
+        y = FT(Inf)
+        x_history = init_history(soltype, FT)
+        y_history = init_history(soltype, FT)
+        return SolutionResults(soltype, x0, false, y, 0, x_history, y_history)
+    end
+
+    y0 = f(x0)
+    y1 = f(x1)
+    
+    x_history = init_history(soltype, x0)
+    y_history = init_history(soltype, y0)
+    
+    for i in 1:maxiters
+        Δx = x1 - x0
+        Δy = y1 - y0
+        
+        if abs(Δy) <= 100 * eps(y1)
+            # Exiting because the function is flat. This is a stall.
+            # The method has only "converged" if the residual is already small.
+            converged = abs(y1) < default_tol(FT).tol # Check for near-zero residual
+            return SolutionResults(soltype, x1, converged, y1, i, x_history, y_history)
+        end
+        
+        # Update x0, y0 to the "previous" state for the next iteration
+        # and for the tolerance check below.
+        x0, y0 = x1, y1
+        
+        # Update x1 in-place using the now-stored previous state in y0.
+        x1 -= y0 * Δx / Δy
+        y1 = f(x1)
+        
+        push_history!(x_history, x1, soltype)
+        push_history!(y_history, y1, soltype)
+
+        # Check for convergence
+        if tol(x0, x1, y1) 
+            return SolutionResults(soltype, x1, true, y1, i, x_history, y_history)
+        end
+    end
+    
+    return SolutionResults(soltype, x1, false, y1, maxiters, x_history, y_history)
+end
+
+
+# Helper: Main iteration loop for Newton methods
```

Comment:
I would change this to a proper docstring and add it to developer docs.

Reply:
I prefer to leave it so we do not duplicate information in the docs.